homie-influx/src/main.rs

```diff
@@ -0,0 +1,388 @@
+        .unwrap_or(DEFAULT_MQTT_PORT);
+
+    let mut mqtt_options = MqttOptions::new(client_name, mqtt_host, mqtt_port);
+    mqtt_options.set_keep_alive(5);
+
+    let mqtt_username = std::env::var("MQTT_USERNAME").ok();
+    let mqtt_password = std::env::var("MQTT_PASSWORD").ok();
+    if let (Some(username), Some(password)) = (mqtt_username, mqtt_password) {
+        mqtt_options.set_credentials(username, password);
+    }
+
+    if std::env::var("MQTT_USE_TLS").is_ok() {
+        let mut client_config = ClientConfig::new();
+        client_config.root_store = rustls_native_certs::load_native_certs()
+            .expect("Failed to load platform certificates.");
+        mqtt_options.set_tls_client_config(Arc::new(client_config));
+    }
+
+    mqtt_options
+}
+
+fn spawn_homie_poll_loop(
+    mut event_loop: HomieEventLoop,
+    controller: Arc<HomieController>,
+    influx_db_client: Client,
+) -> JoinHandle<Result<(), PollError>> {
+    task::spawn(async move {
+        loop {
+            if let Some(event) = controller.poll(&mut event_loop).await? {
+                match event {
+                    Event::PropertyValueChanged {
+                        device_id,
+                        node_id,
+                        property_id,
+                        value,
+                        fresh,
+                    } => {
+                        log::trace!(
+                            "{}/{}/{} = {} ({})",
+                            device_id,
+                            node_id,
+                            property_id,
+                            value,
+                            fresh
+                        );
+                        if fresh {
+                            send_property_value(
+                                controller.as_ref(),
+                                &influx_db_client,
+                                device_id,
+                                node_id,
+                                property_id,
+                            )
+                            .await;
+                        }
+                    }
+                    _ => {
+                        log::info!("Event: {:?}", event);
+                    }
+                }
+            }
+        }
+    })
+}
+
+async fn send_property_value(
+    controller: &HomieController,
+    influx_db_client: &Client,
+    device_id: String,
+    node_id: String,
+    property_id: String,
+) {
+    if let Some(device) = controller.devices().get(&device_id) {
+        if let Some(node) = device.nodes.get(&node_id) {
+            if let Some(property) = node.properties.get(&property_id) {
+                if let Some(point) =
+                    point_for_property_value(device, node, property, SystemTime::now())
+                {
+                    // TODO: What should rp be?
+                    // TODO: Handle errors
+                    influx_db_client
+                        .write_point(point, INFLUXDB_PRECISION, None)
```

Comment:
Seems like it's possible for the influxdb admin to set a default retention policy for a db, so leaving this as None sounds like the best thing to do. https://docs.influxdata.com/influxdb/v1.8/query_language/manage-database/#create-a-default-retention-policy

Reply:
Hopefully that works as expected! Changed the TODO to a comment.